fire/api/firedb.py

```diff
@@ -625,6 +628,141 @@ def basedir_materiale(self):
+            return interval(90001, 99999)
+
+        if fikspunktstype == FikspunktsType.VANDSTANDSBRÆT:
+            raise NotImplementedError(
+                "Fikspunktstypen 'VANDSTANDSBRÆT' er endnu ikke understøttet"
+            )
+
+        raise ValueError("Ukendt fikspunktstype")
+
+    def _opmålingsdistrikt_fra_punktid(self, uuider: List[str]):
+        """
+        Udtræk relevante opmålingsdistrikter, altså dem hvor de adspurgte punkter
+        befinder sig i.
+
+        Hjælpefunktion til tilknyt_landsnumre(). Defineret i seperat funktion
+        med henblik på at kunne mocke den i unit tests.
+        """
+        statement = text(
+            f"""SELECT hs.kode, go.punktid
+                FROM geometriobjekt go
+                JOIN herredsogn hs ON sdo_relate(hs.geometri, go.geometri, 'mask=contains') = 'TRUE'
+                WHERE
+                go.punktid IN ({','.join(uuider)})
+            """
+        )
+
+        return self.session.execute(statement)
+
+    def _løbenumre_i_distrikt(self, distrikt: str):
+        """
+        For et givent opmålingsdistrikt findes alle landsnumre på formen
+        xx-yyy-*****, hvorefter løbenummrene (*****) udskilles og returneres
+        i sorteret orden.
+
+        Hjælpefunktion til tilknyt_landsnumre(). Defineret i seperat funktion
+        med henblik på at kunne mocke den i unit tests.
+        """
+        landsnr = self.hent_punktinformationtype("IDENT:landsnr")
+        sql = text(
+            fr"""SELECT lbnr
+                FROM (
+                    SELECT
+                        regexp_substr(tekst, '.*-.*-(.+)', 1, 1, '', 1) lbnr
+                    FROM punktinfo
+                    WHERE infotypeid={landsnr.infotypeid} AND REGEXP_LIKE(tekst, '{distrikt}-.+$')
+                )
+                ORDER BY lbnr ASC
+                """
+        )
+
+        return [løbenummer for løbenummer in self.session.execute(sql)]
+
+    def tilknyt_landsnumre(
+        self, punkter: List[Punkt], fikspunktstyper: List[FikspunktsType],
+    ) -> List[PunktInformation]:
+        """
+        Tilknytter et landsnummer til punktet hvis der ikke findes et i forvejen.
+
+        Returnerer en liste med IDENT:landsnr PunktInformation'er for alle de fikspunkter i
+        `punkter` som ikke i forvejen har et landsnummer. Hvis alle fikspunkter i `punkter`
+        allerede har et landsnummer returneres en tom liste.
+
+        Kun punkter i Danmark kan tildeles et landsnummer. Det forudsættes at punktet
+        har et tilhørende geometriobjekt og er indlæst i databasen i forvejen.
+        """
+
+        landsnr = self.hent_punktinformationtype("IDENT:landsnr")
+
+        uuider = []
+        punkttyper = {}
+        for punkt, fikspunktstype in zip(punkter, fikspunktstyper):
+            if not punkt.geometri:
+                raise AttributeError("Geometriobjekt ikke tilknyttet Punkt")
+
+            # hvis punktet har et landsnr i forvejen ignoreres det
```

Comment:
"det" henviser til "landsnr", ikke til "Punktet", så kommentaren kommer til at betyde det modsatte af det, der er intentionen.

Jeg mener også at kodeforløbet vil være klarere, hvis du først eliminerer "det ikke forventede", og så foretager "det forventede" på højeste blokniveau, i stedet for nede under en `if` (GI-begrebet "Taoistisk programmering": Eliminer først undtagelserne, og løs så det tilbageværende problem):

```python
for punkt, fikspunktstype in zip(punkter, fikspunktstyper):
    ...
    # Ignorer punkter, der allerede har landsnr
    if landsnr in [pi.infotype for pi in punkt.punktinformationer]:
        continue
    uuider.append(f"'{punkt.id}'")
    punkttyper[punkt.id] = fikspunktstype
```
... det er en detalje uden stor betydning - men hvis du alligevel roder i kommentarlinjen ville det vel være minimalt besvær også at vende `if`-testen om